src/libasr/codegen/asr_to_cpp.cpp

```diff
@@ -518,8 +519,6 @@ Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
             }
             src = fn_name + "(" + args + ")";
         }
-        last_unary_plus = false;
-        last_binary_plus = false;
```

Comment:
`FunctionCall` must set precedence. I don't see it set in the visitor itself. Is it somehow implicitly set elsewhere? I don't know of a good way to ensure that it is set in every `expr` visitor, but for now a good way is to literally explicitly assign to `last_expr_precedence` in every single visitor, or add a comment why that is not needed in the source code. So since I don't see it being assigned, I think it might be a bug? Let's either assign or add a comment here why not.